aws/elasticache/elasticache.tf

```diff
@@ -86,15 +86,17 @@ resource "aws_elasticache_replication_group" "cluster_mode" {
 }
 
 locals {
-  # the format is different for different major Redis versions, see: https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateCacheParameterGroup.html
-  family = "redis${var.major_version == 6 ? "6.x" : "7"}"
-
+  family = var.engine == "valkey" ? (
+    var.major_version == 7 ? "valkey7" : "valkey8"
+  ) : (
+    var.major_version == 6 ? "redis6.x" : "redis7"
+  )
 }
 
 resource "aws_elasticache_parameter_group" "main" {
   count = var.parameter_group_name == null ? 1 : 0
 
-  name   = local.cluster_name
+  name   = "${local.cluster_name}-${var.engine}"
```

Comment:
Is this required for the migration? No other side-effects from a name change here I assume?